src/backend/distributed/commands/table.c

```diff
@@ -1390,3 +1450,52 @@ ErrorIfUnsupportedAlterAddConstraintStmt(AlterTableStmt *alterTableStatement)
 								 colocationId);
 	relation_close(relation, NoLock);
 }
+
+
+/*
+ * AlterTableSchemaStmtObjectAddress returns the ObjectAddress of the table that is the
+ * object of the AlterObjectSchemaStmt.
+ *
+ * This could be called both before or after it has been applied locally. It will look in
+ * the old schema first, if the table cannot be found in that schema it will look in the
+ * new schema. Errors if missing_ok is false and the table cannot be found in either of the
+ * schemas.
+ */
+ObjectAddress
+AlterTableSchemaStmtObjectAddress(Node *node, bool missing_ok)
+{
+	AlterObjectSchemaStmt *stmt = castNode(AlterObjectSchemaStmt, node);
+	Assert(stmt->objectType == OBJECT_TABLE);
+
+	const char *tableName = stmt->relation->relname;
+	Oid tableOid = InvalidOid;
+	if (stmt->relation->schemaname)
+	{
+		const char *schemaName = stmt->relation->schemaname;
+		Oid schemaOid = get_namespace_oid(schemaName, true);
+		tableOid = get_relname_relid(tableName, schemaOid);
+	}
+	else
+	{
+		tableOid = RelnameGetRelid(stmt->relation->relname);
+	}
+
+	if (tableOid == InvalidOid)
+	{
+		const char *newSchemaName = stmt->newschema;
+		Oid newSchemaOid = get_namespace_oid(newSchemaName, true);
```

Comment:
same here, what kind of error would postgres give if we try to set the schema of a table to a non-existing schema.